### PR TITLE
Add last_reported time to insights repots

### LINKED ISSUE #1990: Quipucords insights report should include a last_reported fact in the QPC namespace
## Feature description:
The tally service would like a `last_reported` fact in the qpc namespace.  QPC should generate the `last_reported` at the time the fingerprint scanner is run.  `last_reported` should be included in the qpc namespace facts.

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #1990

### CODE REVIEW

File: quipucords/fingerprinter/tests_fingerprint_task.py [hunk 1]
```diff
@@ -1149,8 +1149,10 @@ def test_process_details_report_success(self):
                     {
                         'namespace': 'qpc',
                         'facts': {
+                            'last_reported': None,
                             'rh_product_certs': [],
-                            'rh_products_installed': []
+                            'rh_products_installed': [],
+
```
Comment (abaiken) [inline quipucords/fingerprinter/tests_fingerprint_task.py]: can we remove this extra line

File: quipucords/fingerprinter/task.py [hunk 2]
```diff
@@ -393,9 +398,8 @@ def _process_details_report(self, manager_interrupt, details_report):
 
                         facts = {'namespace': 'qpc', 'facts': nested_facts}
                         insights_host['facts'] = [facts]
-                        if bool(nested_facts):
-                            insights_host['system_profile'] = \
-                                self.format_system_profile(fingerprint_dict)
+                        insights_host['system_profile'] = \
+                            self.format_system_profile(fingerprint_dict)
```
Comment (abaiken) [inline quipucords/fingerprinter/task.py]: What happens if `self.format_system_profile(fingerprint_dict)` returns an empty dictionary? would assigning `system_profile={}` in our host be accepted by the inventory?

File: quipucords/fingerprinter/tests_fingerprint_task.py (hunk 1 above)
Comment (kholdaway) [reply quipucords/fingerprinter/tests_fingerprint_task.py]: Fixed and swagger updated.

File: quipucords/fingerprinter/task.py (hunk 2 above)
Comment (kholdaway) [reply quipucords/fingerprinter/task.py]: I added code to filter an empty system_profile

**Pull request merged**